storage/engine/mvcc.go

```diff
@@ -957,12 +974,25 @@ func MVCCIterate(engine Engine, key, endKey proto.Key, timestamp proto.Timestamp
 		}
 		value, err := mvccGetInternal(engine, key, metaKey, timestamp, consistent, txn, getValue, buf)
 		if err != nil {
-			return err
+			switch t := err.(type) {
+			case *proto.WriteIntentError:
+				// In the case of WriteIntentErrors, accumulate affected keys but continue scan.
+				if wiErr == nil {
+					wiErr = t
+				} else {
+					wiErr.(*proto.WriteIntentError).Intents = append(wiErr.(*proto.WriteIntentError).Intents, t.Intents...)
```

Comment:
no interface conversion needed if you just start out with `wiErr` as suggested above.